scripts/build_biomass_potentials.py

```diff
@@ -211,14 +211,182 @@ def convert_nuts2_to_regions(bio_nuts2, regions):
+
+    # Renaming some indices
+    index_rename = {
+        "Households": "Residential",
+        "Commercial & public services": "Services",
+        "Domestic navigation": "Domestic Navigation",
+        "International maritime bunkers": "Bunkers",
+    }
+    columns_rename = {"Total": "Total all products", "UK": "GB"}
+    df.rename(index=index_rename, columns=columns_rename, inplace=True)
+    df.sort_index(inplace=True)
+    df.index.names = [None] * len(df.index.names)
+
+    # convert to MWh/a from ktoe/a
+    df *= 11.63 * 1e3
+
+    return df
+
+
+def add_unsustainable_potentials(df):
+    """
+    Add unsustainable biomass potentials to the given dataframe. The difference
+    between the data of JRC and Eurostat is assumed to be unsustainable
+    biomass.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The dataframe with sustainable biomass potentials.
+    unsustainable_biomass : str
+        Path to the file with unsustainable biomass potentials.
+
+    Returns
+    -------
+    pd.DataFrame
+        The dataframe with added unsustainable biomass potentials.
+    """
+    if "GB" in snakemake.config["countries"]:
+        latest_year = 2019
+    else:
+        latest_year = 2021
+    idees_rename = {"GR": "EL", "GB": "UK"}
+    df_unsustainable = (
+        build_eurostat(
+            countries=snakemake.config["countries"],
+            year=max(min(latest_year, investment_year), 1990),
+            input_eurostat=snakemake.input.eurostat,
+            idees_rename=idees_rename,
+        )
+        .xs("Primary production", level=2)
+        .droplevel([1, 2, 3])
+    )
+
+    df_unsustainable.index = df_unsustainable.index.str.strip()
+    df_unsustainable = df_unsustainable.rename(
+        {v: k for k, v in idees_rename.items()}, axis=0
+    )
+
+    bio_carriers = [
+        "Primary solid biofuels",
+        "Biogases",
+        "Renewable municipal waste",
+        "Pure biogasoline",
+        "Blended biogasoline",
+        "Pure biodiesels",
+        "Blended biodiesels",
+        "Pure bio jet kerosene",
+        "Blended bio jet kerosene",
+        "Other liquid biofuels",
+    ]
+
+    df_unsustainable = df_unsustainable[bio_carriers]
+
+    # Phase out unsustainable biomass potentials linearly from 2020 to 2035 while phasing in sustainable potentials
+    reduction_factor = max(
+        0,
+        min(
+            1,
+            1
+            - (investment_year - 2020)
+            / (snakemake.config["biomass"]["unsustainable_phase_out"] - 2020),
+        ),
+    )
+
+    df_wo_ch = df.drop(df.filter(regex="CH\d", axis=0).index)
+
+    df_wo_ch["unsustainable solid biomass"] = (
+        (
+            df_wo_ch.apply(
+                lambda c: c.sum()
+                / df_wo_ch.loc[df_wo_ch.index.str[:2] == c.name[:2]].sum().sum()
+                * df_unsustainable.loc[c.name[:2], "Primary solid biofuels"],
+                axis=1,
+            )
```

Comment:
perhaps move the lambda functions (here and the two other cases) into local functions above to reduce how nested the code is.